lara-typescript/src/interactive-api-client/api.ts

```diff
@@ -388,3 +391,67 @@ export const getInteractiveSnapshot = (options: IGetInteractiveSnapshotOptions)
 export const getLibraryInteractiveList = (options: IGetLibraryInteractiveListRequest) => {
   THROW_NOT_IMPLEMENTED_YET("getLibraryInteractiveList");
 };
+
+export const writeAttachment = (params: Omit<IWriteAttachmentRequest, "requestId" | "operation">) => {
+  return new Promise<void>((resolve, reject) => {
+    const client = getClient();
+    const { name, content, options } = params;
+    const request: IAttachmentUrlRequest = { name, operation: "write", requestId: client.getNextRequestId() };
+    client.addListener("attachmentUrl", async (response: IAttachmentUrlResponse) => {
```

Comment:
I think this needs to be only listening for responses that have the matching requestId. I can't remember if there is some existing addListener mechanism to make this easier.

The reason for this is that the interactive might add multiple attachments and the responses could come out of order.

Reply:
The `requestId` is passed after the callback in line 413.